### PR TITLE
Changes SpinBox for a TextField

### LINKED ISSUE #542: GzSpinBox causes the Qt Scene Graph to crash on macOS
## Environment
* OS Version: macOS, `qt@5 5.15.8_3`
* Source or binary build? either

## Description
* Expected behavior: GUI plugins that use GzSpinBox should work correctly.
* Actual behavior: The GUI application crashes. The error occurs on the Qt scene graph render thread.

## Steps to reproduce

1.  Update and upgrade `brew`. This will install `qt@5 5.15.8_3`. 
2. Run `gz sim -g shapes.sdf`
3. Open the Component Inspector (if not already opened)

## Output

See https://github.com/gazebosim/gz-sim/issues/1987.

The problem appears to be a conflict between QtQuickControls 1.4 and QtQuickControls 2.2. Both versions have a SpinBox but the behaviour is very different. The version in QtQuickControls 2.2 is for integer values only.

The QML documentation suggests an approach to use the controls for floats, but it is badly flawed and will not work for the use cases needed in `gz-sim`. A reimplementation of a spin box for doubles is needed. Docs and discussion:

-  https://doc.qt.io/qt-5/qml-qtquick-controls2-spinbox.html#validator-prop
- https://stackoverflow.com/questions/43406830/how-to-use-float-in-a-qml-spinbox

This is a serious bug on macOS as it causes an application crash on start up for any of the default examples (because the default `gui.config` includes the component inspector which indirectly loads the spin box). 

## Possible approach

One of the answers in the SO article suggests a replacement SpinBox supporting doubles. This is almost a drop-in replacement for GzSpinBox. The branches below make the modifications to `gz-gui7` and `gz-sim7`:

- https://github.com/srmainwaring/gz-gui/tree/wips/7/wip-doublespinbox
- https://github.com/srmainwaring/gz-sim/tree/wips/7/wip-doublespinbox

### PR DESCRIPTION
**Pull request by Voldivh**
# 🦟 Bug fix

Fixes #542 

## Summary
Changes the implementation of GzSpinBox from a SpinBox to a TextField for MacOS.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/gz/gui/qml/GzSpinBox.qml [hunk 1]
```diff
@@ -18,12 +18,31 @@ import QtQuick 2.9
 import QtQuick.Controls 1.4
 import QtQuick.Controls.Styles 1.4
```
Comment (azeey) [inline include/gz/gui/qml/GzSpinBox.qml:19]: For this, we should use the newer version of QtQuick Controls and remove the next line.

```suggestion
import QtQuick.Controls 2.15
import QtQuick.Controls.Styles 2.15
```

File: include/gz/gui/qml/GzSpinBox.qml [hunk 2]
```diff
@@ -18,12 +18,31 @@ import QtQuick 2.9
...
+    onEditingFinished: {
+          parent.value = parseFloat(text)
+    } 
+    style: TextFieldStyle{
+      background: Rectangle {
+        implicitWidth: 70
+        implicitHeight: 40
+        border.color: "gray"
+      }
     }
```
Comment (azeey) [inline include/gz/gui/qml/GzSpinBox.qml:51]: The `TextField` from 2.15, doesn't have style. We'd just use `background` instead:
```suggestion
    background: Rectangle {
      implicitWidth: 70
      implicitHeight: 40
      border.color: "gray"  
    }
```

Comment (Voldivh) [reply include/gz/gui/qml/GzSpinBox.qml]: Done.

File: include/gz/gui/qml/GzSpinBox.qml (hunk 1 above)
Comment (Voldivh) [reply include/gz/gui/qml/GzSpinBox.qml]: Done.

**Comment by azeey** [comment]
Closing since https://github.com/gazebosim/gz-gui/issues/542 is no longer an issue.

**Pull request closed without merging**